backend/src/database/class_enrollments.go

```diff
@@ -215,10 +218,53 @@ func (db *DB) UpdateProgramClasses(ctx context.Context, classIDs []int, classMap
 				Error; err != nil {
 				tx.Rollback()
 				return newUpdateDBError(err, "class enrollment statuses")
+                                                                                                                                    			}
+		}
+
+		// truncate recurrence rule and cancel future overrides
+		for _, classID := range classIDs {
+			var class models.ProgramClass
+			if err := tx.Preload("Facility").Preload("Events", func(db *gorm.DB) *gorm.DB { return db.Order("created_at desc") }).First(&class, "id = ?", classID).Error; err != nil {
+				tx.Rollback()
+				return newNotFoundDBError(err, "program classes")
+			}
+			if len(class.Events) > 0 {
+				active := class.Events[0]
+				loc, _ := time.LoadLocation(class.Facility.Timezone)
+				cutoffDate := time.Now().In(loc)
+				until := cutoffDate.Format("20060102") + "T000000Z"
+				updatedRule := models.ReplaceOrAddUntilDate(active.RecurrenceRule, until)
+				if updatedRule != active.RecurrenceRule {
+					if err := tx.Model(&models.ProgramClassEvent{}).Where("id = ?", active.ID).Update("recurrence_rule", updatedRule).Error; err != nil {
+						tx.Rollback()
+						return newUpdateDBError(err, "program_class_events")
+					}
+					cle := models.NewChangeLogEntry("program_classes", "event_rescheduled_series", models.StringPtr(active.RecurrenceRule), &updatedRule, uint(classID), classMap["update_user_id"].(uint))
```

Comment:
@JamesUnlocked See my comment on the other file about not needing to record the changelog entry when we cancel/complete a class.